### ISSUE TITLE
[test-framework] Add mTLS support for local environment

### ISSUE THREAD

**Issue by nmittler** [istio/istio#8350]
All communication for local testing is currently plaintext.  We need to support mTLS within the local mesh.

**Comment by hklai** [comment]
@JimmyCYJ could you help with this?

**Comment by ozevren** [comment]
@JimmyCYJ is this something that you will/can work for Code Mauve/Release 1.2?

**Comment by JimmyCYJ** [comment]
This depends on a stand-alone Citadel https://github.com/istio/istio/issues/9177, let me check with @myidpt

**Comment by pitlv2109** [comment]
Any update on this work?

**Comment by nmittler** [comment]
@pitlv2109 I'm working on a rewrite of the native echo component using docker (#14614). Once that lands, this issue should be done (or very close).

**Comment by myidpt** [comment]
Downgrading it to P1. Jimmy is working on writing tests to increase the test coverage now.

**Comment by howardjohn** [comment]
Duplicate of https://github.com/istio/istio/issues/13778
